File: python/test_algo/ensemble.py
from __future__ import annotations

from openjiuwen.algorithm_provider import AlgorithmProvider
from .passthrough import Passthrough


class Ensemble(AlgorithmProvider):
    name = "ensemble"

    def decide(self, request, ctx):
        inner = Passthrough()
        decision = inner.decide(request, ctx)
        decision["reasoning"] = "ensemble: stub, falls back to first target"
        return decision

File: crates/algorithms/src/test_algo/test_algorithm/ensemble.rs
use openjiuwen_protocol::{Decision, RouteRequest, RouterError};

use crate::test_algo::{AlgorithmProvider, RouteContext};

/// 集成/混合。骨架阶段退化为直通。
pub struct Ensemble;

impl AlgorithmProvider for Ensemble {
    fn name(&self) -> &str {
        "ensemble"
    }

    fn decide(&self, request: &RouteRequest, ctx: &RouteContext) -> Result<Decision, RouterError> {
        let available = ctx.targets.without(&request.exclusions);
        let model = available.first().ok_or(RouterError::NoTarget)?;
        Ok(Decision::answer(
            model,
            "ensemble: stub, falls back to first target",
        ))
    }
}
